### PR TITLE
Use Docker to test external storages 

### LINKED ISSUE #226: Rewrite tests to use docker
Like aiopg
We need to use dockerized redis and memcache servers in our tests.
It simplifies local development.

### PR DESCRIPTION
**Pull request by panagiks**
Fixes #226.

Both memcached and redis are now tested using Docker.

Having said that there were a couple of things I couldn't get to work:
* Using aiodocker (as @jettify suggested), sorry :s I also liked the idea of using aiodocker
* Using tags for the containers (now the latest image is pulled), pytest wasn't recognizing the `redis_tag` (that I had created using aiopg tests as an example) as a pytest fixture.

I believe that both can be done in the future as an enhancement, but the decision is up to you.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import aioredis
+import asyncio
+import gc
+import pytest
+import time
+import uuid
+from docker import Client as DockerClient
+
+
+@pytest.yield_fixture(scope='session')
```
Comment (asvetlov) [inline tests/conftest.py]: I pretty sure `yield_fixture` could be replaced by just `@pytest.fixture` in modern pytest.
We still have `yield_fixture` in very many *aio-libs* projects.
Need for a hero for replace them all or ending of deprecation period for `@pytest.fixture` :)

Comment (panagiks) [reply tests/conftest.py]: I can go through a few of them (probably next weekend) :)

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        loop.call_soon(loop.stop)
+        loop.run_forever()
+        loop.close()
+    gc.collect()
+    asyncio.set_event_loop(None)
+
+
+@pytest.fixture(scope='session')
+def session_id():
+    '''Unique session identifier, random string.'''
```
Comment (jettify) [inline tests/conftest.py]: Very minor thing, lets use double quotes here.

Comment (panagiks) [reply tests/conftest.py]: Sure :)

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        detach=True,
+    )
+    docker.start(container=container['Id'])
+    inspection = docker.inspect_container(container['Id'])
+    host = inspection['NetworkSettings']['IPAddress']
+    delay = 0.001
+    for i in range(100):
+        try:
+            conn = loop.run_until_complete(
+                aioredis.create_connection((host, 6379))
```
Comment (jettify) [inline tests/conftest.py]: in order to fix 
```
RuntimeError: There is no current event loop in thread 'MainThread'.
```
you need to pass loop explicitly here

**Comment by jettify** [comment]
PR lock good to me, we just need to make it green

File: tests/conftest.py (hunk 3 above)
Comment (panagiks) [reply tests/conftest.py]: To avoid multiple pushes, you mean pass it to `run_until_complete` as a kwarg right?

Comment (panagiks) [reply tests/conftest.py]: (retracted run_until_complete doesn't take loop kwarg :s so I guess in create_connection)

**Comment by panagiks** [comment]
Should be ok now :)

**Comment by jettify** [comment]
@panagiks thanks a lot!

**Comment by asvetlov** [comment]
Awesome work!

**Comment by panagiks** [comment]
Thank you :)

**Pull request merged**